### PR TITLE
25 sign up screen UI 2

### PR DESCRIPTION
**Pull request by Njumbi**

### CODE REVIEW

File: app/src/main/java/com/todoist_android/ui/auth/SignupFragment.kt [hunk 1]
```diff
@@ -1,9 +1,111 @@
...
+        super.onViewCreated(view, savedInstanceState)
+
+        viewLifecycleOwner.lifecycleScope.launchWhenStarted {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED){
+                viewModel.signupResponse.collect {
+                    when(it){
+                        is APIResource.Success->{
+                            view.findNavController().navigate(R.id.action_signupFragment_to_loginFragment)
+                            binding.progressbarTwo.visibility = View.GONE
+                            Toast.makeText(requireContext(),it.toString(),Toast.LENGTH_SHORT).show()
```
Comment (malcolmmaima) [inline app/src/main/java/com/todoist_android/ui/auth/SignupFragment.kt:46]: `Toast.makeText(requireContext(),"Signup successful! Please login",Toast.LENGTH_LONG).show()`

File: app/src/main/java/com/todoist_android/ui/auth/SignupFragment.kt [hunk 2]
```diff
@@ -1,9 +1,111 @@
...
+            }
+            if (userName.trim().length<= 2){
+                binding.etUsername.error ="Enter a Name with more than two words"
+                return@setOnClickListener
+            }
+            if (binding.etEmail.text.isNullOrEmpty()){
+                binding.etEmail.error ="Please Enter your Email"
+                return@setOnClickListener
+            }
+            if (!validateEmail(binding.etEmail.text.toString())){
```
Comment (malcolmmaima) [inline app/src/main/java/com/todoist_android/ui/auth/SignupFragment.kt:81]: `!validateEmail(binding.etEmail.text.toString().trim())`

noticed even after valid email, if there's a space at the end it'll bring up the error

File: app/src/main/java/com/todoist_android/ui/auth/SignupFragment.kt [hunk 3]
```diff
@@ -1,9 +1,111 @@
...
+
+        binding.progressbarTwo.visibility = View.GONE
+        binding.btnSignup.setOnClickListener {
+
+            binding.progressbarTwo.visibility = View.VISIBLE
+
+            val userName = binding.etUsername.text.toString()
+            val userEmail = binding.etEmail.text.toString()
+            val userPassword = binding.etPassword.text.toString()
+            val confirmPassword = binding.etConfirmPassword.text.toString()
```
Comment (malcolmmaima) [inline app/src/main/java/com/todoist_android/ui/auth/SignupFragment.kt:66]: `val userName = binding.etUsername.text.toString().trim()
            val userEmail = binding.etEmail.text.toString().trim()
            val userPassword = binding.etPassword.text.toString().trim()
            val confirmPassword = binding.etConfirmPassword.text.toString().trim()`

**Review by malcolmmaima — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/todoist_android/ui/auth/SignupFragment.kt (hunk 1 above)
Comment (Njumbi) [reply app/src/main/java/com/todoist_android/ui/auth/SignupFragment.kt]: Let me add this

**Review by malcolmmaima — APPROVED** [review]

**Comment by malcolmmaima** [comment]
[Sign Up Screen -Android](https://trello.com/c/z41mVux0/25-sign-up-screen-android)

**Pull request merged**